### ISSUE TITLE
Remove `pre` from the main action and move it to azure/login/clear-credentials

### ISSUE THREAD

**Issue by jsoref** [Azure/login#474]
#431  is considering reworking things a bit.

I propose kicking `pre` out of the [main action](https://github.com/Azure/login/blob/e1a0ade0c10e6fce656e01e74e6ca3c5054a0907/action.yml#L40-L44):

`action.yml`
```diff
...
 runs: 
   using: 'node20' 
-  pre: 'lib/cleanup/index.js' 
   main: 'lib/main/index.js' 
   post: 'lib/cleanup/index.js' 
```

and create `clear-credentials/action.yml`:
```yaml
name: 'Azure Login: Clear Credentials'
description: 'Clear Azure credentials (use this once before your first `uses: azure/login` step)'
branding:
  icon: 'login.svg'
  color: 'blue'
runs:
   main: 'lib/cleanup/index.js' 
```

Note: I'm not sure that `login.svg` is doing anything here. https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#brandingicon lists the valid values and `login.svg` isn't on the list (nor is there a `login.svg` in the repository). (There doesn't appear to be documentation on this.)

This would be a breaking change, so you'd bump the major version of `azure/login`...

This would mean that workflows such as https://github.com/microsoft/playwright/blob/b8546eb35ea23c735be656b5675988cbc5486c7b/.github/workflows/tests_primary.yml that use local actions as noted in https://github.com/actions/runner/issues/3397 would stop triggering dozens of warnings they can't do anything about.

But it would still allow people concerned about credentials that resulted in #384 to be able to clear credentials at start and have more than one login step.

**Comment by olivierlefloch** [comment]
I second this ; our use case is we're using `azure/login` in an internal re-usable, configurable `setup` workflow which does not always login with Azure. So the `azure/login` invocation is conditionally enabled:

```
    - name: "Azure login"
      if: inputs.auth-with-azure == 'true'
      uses: azure/login@v2
      …
```

However, the `pre` step is run in all cases, which incurs overhead for all our jobs.

**Comment by github-fetch-packages** [comment]
Would be amazing to not require the pre run step. It incurs overhead for all jobs; regardless of whether or not azure/login is actually used.

**Comment by YanaXu** [comment]
Hi @jsoref  and @olivierlefloch, we've released a new version [v2.2.0](https://github.com/Azure/login/releases/tag/v2.2.0). In this version, we removed the `pre` step and add 2 env variables to enable/disable pre or post cleanup.
`v2` is aligned with `v2.2.0` now.
Please try the new version. You can also refer to [Enable/Disable the cleanup steps](https://github.com/Azure/login/tree/v2.2.0?tab=readme-ov-file#enabledisable-the-cleanup-steps).

**Comment by jsoref** [comment]
Thanks @YanaXu, I believe this addresses my concerns.
